flume/src/test/java/org/apache/apex/malhar/flume/sink/FlumeSinkTest.java

```diff
@@ -54,12 +54,14 @@ public void testServer() throws InterruptedException, IOException
       @Override
       public synchronized void unadvertise(Service<byte[]> service)
       {
+        logger.info("Unadvertise invoked");
         notify();
       }
 
       @Override
       public synchronized void advertise(Service<byte[]> service)
       {
+        logger.info("Advertise invoked");
```

Comment:
try adding Thread.sleep(1), also add a comment that this is due to race condition

Reply:
Done.